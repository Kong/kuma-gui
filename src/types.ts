export interface KDSSubscription {
  config: string
  id: string
  globalInstanceId: string
  connectTime: string
  disconnectTime?: string
  status: any
  version: any
}

export interface Zone {
  type: 'Zone'
  name: string
  creationTime: string
  modificationTime: string
  ingress: {
    address: string
  }
}

export interface ZoneInsight {
  subscriptions: KDSSubscription[]
}

export interface ZoneOverview {
  type: 'ZoneOverview'
  mesh: string
  name: string
  creationTime: string
  modificationTime: string
  zone: Zone
  zoneInsight: ZoneInsight
}

export interface ResourceResponse<T = {}> {
  total: number
  items: T[]
  next: boolean | null
}

export interface AllResourceResponse<T = {}> {
  data: T[]
  total: number
}
