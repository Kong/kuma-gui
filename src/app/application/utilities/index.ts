type URLParamDefault = string | number | boolean
type URLParamValue = string | null

const difference = <T>(a: T[], b: T[]): T[] => a.filter(item => !b.includes(item))

const includes = <T extends readonly string[]>(arr: T, item: string): item is T[number] => {
  return arr.includes(item as T[number])
}

const createUniqueId = (j = 0) => {
  let i = j
  return (prefix = 'unique') => {
    i++
    return `${prefix}-${i}`
  }
}

export const uniqueId = createUniqueId()

export const beforePaint = <T extends (...args: any[]) => void>(fn: T) => {
  let num: number
  return (...args: Parameters<T>) => {
    if (num) {
      window.cancelAnimationFrame(num)
    }
    num = window.requestAnimationFrame(fn.bind(fn, ...args))
  }
}

const supportedAttrs = ['class'] as const
type SupportedAttrs = typeof supportedAttrs[number]
export const createAttrsSetter = ($el = document.documentElement) => {
  if (!$el) {
    return () => {}
  }
  const originalClasses = [...$el.classList]
  return beforePaint((attrs: Partial<Record<SupportedAttrs, string>>[]) => {
    const flat = attrs.reduce<Record<SupportedAttrs, string[]>>((prev, item) => {
      return Object.entries(item).reduce(
        (prev, [key, value]) => {
          if (includes(supportedAttrs, key) && value) {
            prev[key].push(value)
          }
          return prev
        }, prev,
      )
    }, { class: [] })

    // omit any classes that were on the node previous to our application starting
    const currentClasses = difference([...$el.classList], originalClasses)
    // anything in currentClasses that isn't in our tree of attrs, remove
    $el.classList.remove(...difference(currentClasses, flat.class))
    // anything in our tree of attrs that isn't in currentClasses, add
    $el.classList.add(...difference(flat.class, currentClasses))
  })
}

// normalizes url params from "value or array of values" to value
// if the url params is an array we use the first value
export const urlParam = function <T extends URLParamValue> (param: T | T[]): T {
  return (Array.isArray(param) ? param[0] : param)
}

//
export const normalizeUrlParam = (param: URLParamValue, def: URLParamDefault): URLParamDefault => {
  switch (true) {
    case typeof def === 'boolean':
      return param === null ? true : def
    case typeof def === 'number': {
      const value = param === null || param.length === 0 ? def : Number(decodeURIComponent(param))
      if (isNaN(value)) {
        return Number(def)
      } else {
        return value
      }
    }
    case typeof def === 'string': {
      return param === null || param.length === 0 ? def : decodeURIComponent(param)
    }
  }
  throw new TypeError('URL parameters can only be string | number | boolean')
}
export const createTitleSetter = ($doc = document) => {
  return beforePaint((title) => {
    $doc.title = title
  })
}
// when used with router.push, prevents things like `q=` (i.e. key= but with no value)
export const cleanQuery = <T extends Record<string, unknown>>(params: Record<string, string | boolean | number | undefined>, originalQuery: T) => {
  const query = {
    ...originalQuery as Record<string, string | undefined | null>,
  }
  const processed = Object.entries(params).reduce((prev, [key, value]) => {
    // Only add parameters for non-empty strings or non-false boolean
    switch (true) {
      case value === true:
        // if the value is strictly true that means we use shorthand boolean true
        // i.e. `?prop`
        prev[key] = null
        break
      case value !== false && String(value).length > 0:
        // if we aren't strictly false (which will use the default)
        // convert to a string and check the length
        // anything greater than zero encode and use
        prev[key] = encodeURIComponent(String(value))
        break
      default:
        // false, or any zero length value unset entirely
        prev[key] = undefined
    }
    return prev
  }, query)
  return {
    ...query,
    ...processed,
  }
}
