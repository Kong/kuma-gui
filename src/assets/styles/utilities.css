/* Utilities */

.kutil-sr-only {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
}

.kutil-img-full {
  max-width: 100%;
  width: 100%;
  height: auto;
}

.kutil-img-fluid {
  max-width: 100%;
  height: auto;
}

.kutil-clear-text {
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
}

.kutil-button-base {
  font-family: $base-font-family;
  font-weight: $fw-medium;
  color: #fff !important;
  background-color: $green-base;
  padding: 0.563rem 2.25rem;
  border-radius: 3px;
  box-shadow: $base-soft-shadow;
  text-align: center;
  text-shadow: 0 2px 0 $green-dark;
  box-sizing: border-box;
  cursor: pointer;
}

.kutil-button-base:hover,
.kutil-button-base:active,
.kutil-button-base:focus,
.kutil-button-base:visited {
  color: #fff !important;
}

.kutil-button-base:hover {
  /* background-color: lighten($green-base, 3.5); */
}

.kutil-quotify {
  quotes: "“" "”" "‘" "’";
}

.kutil-quotify:before {
  content: open-quote;
}

.kutil-quotify:after {
  content: close-quote;
}

/* @mixin retina-mq {
  @media (-webkit-min-device-pixel-ratio: 2), (min-resolution: 192dpi) {
    @content;
  }
}

@mixin target-ie {
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    @content;
  }
} */

.kutil-sticky {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
}

.kutil-sticky:before,
.kutil-sticky:after {
  content: "";
  display: table;
}

/* New for Kuma */

.arrow-link {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: var(--blue-4);
}

.k-button {
  font-weight: normal !important;
  display: flex;
  align-items: center;
}

.k-button:after,
.arrow-link:after {
  display: inline-block;
  content: "›";
  margin-left: 5px;
}

.capitalize {
  display: inline-block;
  text-transform: capitalize;
}
