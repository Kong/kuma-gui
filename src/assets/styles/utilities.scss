/* Utilities */

.kutil-sr-only {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
}

.kutil-img-full {
  max-width: 100%;
  width: 100%;
  height: auto;
}

.kutil-img-fluid {
  max-width: 100%;
  height: auto;
}

.kutil-clear-text {
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
  -moz-osx-font-smoothing: grayscale;
}

.kutil-quotify {
  quotes: "“" "”" "‘" "’";
}

.kutil-quotify:before {
  content: open-quote;
}

.kutil-quotify:after {
  content: close-quote;
}

/* @mixin retina-mq {
  @media (-webkit-min-device-pixel-ratio: 2), (min-resolution: 192dpi) {
    @content;
  }
}

@mixin target-ie {
  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    @content;
  }
} */

.kutil-sticky {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
}

.kutil-sticky:before,
.kutil-sticky:after {
  content: "";
  display: table;
}

/* New for Kuma */

.capitalize {
  display: inline-block;
  text-transform: capitalize;
}

/* position */
.fixed-top {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1030;
}

.text-left {
  text-align: left;
}

.text-center {
  text-align: center;
}

.text-right {
  text-align: right;
}

.text-uppercase {
  text-transform: uppercase;
}

.text-italic {
  font-style: italic;
}

.font-normal {
  font-weight: 400;
}

.font-light {
  font-weight: 100;
}

.font-semibold {
  font-weight: 500;
}

.font-bold {
  font-weight: 700;
}

/* badges */
.badge {
  display: inline-block;
  padding: 0.25rem 0.4rem;
  font-size: 75%;
  line-height: 1;
  text-align: center;
  white-space: nowrap;
  vertical-align: baseline;
  border-radius: 0.25rem;
}

.top-border {
  padding-top: 1.5rem;
  border-top: 1px solid var(--black-10);
}

.bottom-border {
  padding-bottom: 1.5rem;
  border-bottom: 1px solid var(--black-10);
}

.text-muted {
  color: var(--black-45) !important;
}
.text-danger {
  color: var(--red-500) !important;
}

.h-100 {
  height: 100%;
}
.h-screen {
  min-height: 100vh;
}

code.inline {
  @apply py-1 px-2 m-0 text-sm rounded;

  background-color: var(--gray-1);
}

code.block {
  @apply my-4 rounded-md overflow-auto p-4 text-sm;

  background-color: var(--gray-1);
  border: 1px solid var(--gray-4);
}
