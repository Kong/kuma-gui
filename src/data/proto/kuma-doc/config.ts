/* eslint-disable */

export const protobufPackage = "doc";

export interface Config {
  type: Config_Type;
  name: string;
  fileName: string;
  templateName: string;
}

export enum Config_Type {
  Undefined = "Undefined",
  Policy = "Policy",
  Proxy = "Proxy",
  Other = "Other",
  UNRECOGNIZED = "UNRECOGNIZED",
}

export function config_TypeFromJSON(object: any): Config_Type {
  switch (object) {
    case 0:
    case "Undefined":
      return Config_Type.Undefined;
    case 1:
    case "Policy":
      return Config_Type.Policy;
    case 2:
    case "Proxy":
      return Config_Type.Proxy;
    case 99:
    case "Other":
      return Config_Type.Other;
    case -1:
    case "UNRECOGNIZED":
    default:
      return Config_Type.UNRECOGNIZED;
  }
}

export function config_TypeToJSON(object: Config_Type): string {
  switch (object) {
    case Config_Type.Undefined:
      return "Undefined";
    case Config_Type.Policy:
      return "Policy";
    case Config_Type.Proxy:
      return "Proxy";
    case Config_Type.Other:
      return "Other";
    case Config_Type.UNRECOGNIZED:
    default:
      return "UNRECOGNIZED";
  }
}

function createBaseConfig(): Config {
  return { type: Config_Type.Undefined, name: "", fileName: "", templateName: "" };
}

export const Config = {
  fromJSON(object: any): Config {
    return {
      type: isSet(object.type) ? config_TypeFromJSON(object.type) : Config_Type.Undefined,
      name: isSet(object.name) ? String(object.name) : "",
      fileName: isSet(object.fileName) ? String(object.fileName) : "",
      templateName: isSet(object.templateName) ? String(object.templateName) : "",
    };
  },

  toJSON(message: Config): unknown {
    const obj: any = {};
    message.type !== undefined && (obj.type = config_TypeToJSON(message.type));
    message.name !== undefined && (obj.name = message.name);
    message.fileName !== undefined && (obj.fileName = message.fileName);
    message.templateName !== undefined && (obj.templateName = message.templateName);
    return obj;
  },

  fromPartial(object: DeepPartial<Config>): Config {
    const message = Object.create(createBaseConfig()) as Config;
    message.type = object.type ?? Config_Type.Undefined;
    message.name = object.name ?? "";
    message.fileName = object.fileName ?? "";
    message.templateName = object.templateName ?? "";
    return message;
  },
};

type Builtin = Date | Function | Uint8Array | string | number | boolean | undefined;

export type DeepPartial<T> = T extends Builtin ? T
  : T extends Array<infer U> ? Array<DeepPartial<U>> : T extends ReadonlyArray<infer U> ? ReadonlyArray<DeepPartial<U>>
  : T extends {} ? { [K in keyof T]?: DeepPartial<T[K]> }
  : Partial<T>;

function isSet(value: any): boolean {
  return value !== null && value !== undefined;
}
