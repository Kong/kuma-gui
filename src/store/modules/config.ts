import Kuma from '@/services/kuma'
import { StoreOptions, ActionTree, GetterTree, MutationTree } from 'vuex'
import { RootInterface } from '..'

export interface ConfigInterface {
  status: string | null
  environment: string | null
}

export type ConfigType = StoreOptions<ConfigInterface>

const state: ConfigInterface = {
  status: null,
  environment: null,
}

const mutations: MutationTree<ConfigInterface> = {
  SET_STATUS: (state, status: string) => (state.status = status),
  SET_ENVIRONMENT: (state, value) => (state.environment = value),
}

const getters: GetterTree<ConfigInterface, RootInterface> = {
  getStatus: (state) => state.status,
  getEnvironment: (state) => state.environment,
}

const actions:(api: Kuma) => ActionTree<ConfigInterface, RootInterface> = (api: Kuma) => ({
  // get the status of the API
  getStatus ({ commit }) {
    return api.getStatus()
      .then(response => {
        commit('SET_STATUS', response)
      })
  },
})

export default (api: Kuma) => ({
  namespaced: true,
  state,
  getters,
  mutations,
  actions: actions(api)
})
