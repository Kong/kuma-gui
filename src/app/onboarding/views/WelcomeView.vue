<template>
  <RouteView>
    <RouteTitle
      :title="t('onboarding.routes.welcome.title', {name: t('common.product.name')})"
    />
    <AppView>
      <div>
        <OnboardingPage>
          <template #header>
            <OnboardingHeading>
              <template #title>
                Welcome to {{ PRODUCT_NAME }}
              </template>

              <template #description>
                <p>
                  Congratulations on downloading {{ PRODUCT_NAME }}! You are just a <strong>few minutes</strong> away from getting your service mesh fully online.
                </p>

                <p>
                  We have automatically detected that you are running on <strong>{{ enviromentFormatted }}</strong>.
                </p>
              </template>
            </OnboardingHeading>
          </template>

          <template #content>
            <h2 class="text-center">
              Letâ€™s get started:
            </h2>

            <div class="item-status-list-wrapper">
              <ul class="item-status-list">
                <li
                  v-for="item in statuses"
                  :key="item.name"
                >
                  <span class="circle mr-2">
                    <KIcon
                      v-if="item.status"
                      icon="check"
                      size="14"
                      color="currentColor"
                    />
                  </span>

                  {{ item.name }}
                </li>
              </ul>
            </div>
          </template>

          <template #navigation>
            <OnboardingNavigation next-step="onboarding-deployment-types" />
          </template>
        </OnboardingPage>

        <WelcomeAnimationSvg :longer="isMulticluster" />
      </div>
    </AppView>
  </RouteView>
</template>

<script lang="ts" setup>
import { KIcon } from '@kong/kongponents'
import { computed } from 'vue'

import OnboardingHeading from '../components/OnboardingHeading.vue'
import OnboardingNavigation from '../components/OnboardingNavigation.vue'
import OnboardingPage from '../components/OnboardingPage.vue'
import WelcomeAnimationSvg from '../components/WelcomeAnimationSvg.vue'
import AppView from '@/app/application/components/app-view/AppView.vue'
import RouteTitle from '@/app/application/components/route-view/RouteTitle.vue'
import RouteView from '@/app/application/components/route-view/RouteView.vue'
import { PRODUCT_NAME } from '@/constants'
import { useStore } from '@/store/store'
import { useI18n } from '@/utilities'

const store = useStore()
const { t } = useI18n()

const enviromentFormatted = computed(() => {
  const environment = store.getters['config/getEnvironment']
  return environment.charAt(0).toUpperCase() + environment.slice(1)
})

const isMulticluster = computed(() => store.getters['config/getMulticlusterStatus'])
const statuses = computed(() => [
  {
    name: `Run ${PRODUCT_NAME} control plane`,
    status: true,
  },
  {
    name: 'Learn about deployments',
    status: false,
  },
  {
    name: 'Learn about configuration storage',
    status: false,
  },
  ...isMulticluster.value ? [{ name: 'Add zones', status: false }] : [],
  {
    name: 'Create the mesh',
    status: false,
  },
  {
    name: 'Add services',
    status: false,
  },
  {
    name: 'Go to the dashboard',
    status: false,
  },
])
</script>

<style lang="scss" scoped>
.item-status-list-wrapper {
  display: flex;
  justify-content: center;
}

.item-status-list {
  margin-top: var(--spacing-md);
}

.item-status-list > * + * {
  margin-top: var(--spacing-xs);
}

.circle {
  height: 1rem;
  width: 1rem;
  border-radius: 50%;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  background-color: var(--grey-300);
}
</style>
