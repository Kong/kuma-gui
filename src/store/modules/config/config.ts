import Kuma from '@/services/kuma'
import { ActionTree, GetterTree, MutationTree } from 'vuex'
import { RootInterface } from '../..'
import { ConfigInterface, ConfigType } from './config.types'

const state: ConfigInterface = {
  status: null,
  tagline: null,
  version: null,
  clientConfig: null
}

const mutations: MutationTree<ConfigInterface> = {
  SET_CONFIG_DATA: (state, config) => (state.clientConfig = config),
  SET_STATUS: (state, status) => (state.status = status),
  SET_TAGLINE: (state, tagline) => (state.tagline = tagline),
  SET_VERSION: (state, version) => (state.version = version),
}

const getters: GetterTree<ConfigInterface, RootInterface> = {
  getStatus: (state) => state.status,
  getConfig: (state) => state.clientConfig,
  getEnvironment: (state) => state.clientConfig?.environment,
  getMode: (state) => state.clientConfig?.mode,
  getTagline: (state) => state.tagline,
  getVersion: (state) => state.version,
  getMulticlusterStatus: (state, getters) => {
    // is Kuma running in Multi-Zone mode?

    let status

    if (process.env.NODE_ENV === 'development' && process.env.VUE_APP_FAKE_MULTIZONE === 'true') {
      status = true

      console.warn(
        '%c âœ¨You are currently faking Multi-Zone mode.',
        'background: black; color: white; display: block; padding: 0.25rem;'
      )
    } else {
      status = getters.getMode === 'global'
    }

    return status
  },
}

const actions:(api: Kuma) => ActionTree<ConfigInterface, RootInterface> = (api: Kuma) => ({

  // get the general Kuma config (this differs from the API config endpoint)
  getConfig ({ commit }) {
    return api.getConfig()
      .then(response => {
        commit('SET_CONFIG_DATA', response)
      })
  },

  // get the status of the API
  getStatus ({ commit }) {
    return api.getStatus()
      .then(response => {
        commit('SET_STATUS', response)
      })
  },

  // get the current tagline
  getTagline ({ commit }) {
    return api.getInfo()
      .then(response => {
        commit('SET_TAGLINE', response.tagline)
      })
      .catch(error => {
        console.error(error)
      })
  },

  // get the current version
  getVersion ({ commit }) {
    return api.getInfo()
      .then(response => {
        commit('SET_VERSION', response.version)
      })
      .catch(error => {
        console.error(error)
      })
  },
})

export default (api: Kuma): ConfigType => ({
  namespaced: true,
  state,
  getters,
  mutations,
  actions: actions(api)
})
