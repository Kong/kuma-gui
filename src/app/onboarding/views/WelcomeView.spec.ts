import { describe, expect, test } from '@jest/globals'
import { mount } from '@vue/test-utils'

import WelcomeView from './WelcomeView.vue'
import { useStore } from '@/utilities'
import { ClientConfigInterface } from '@/store/modules/config/config.types'
import * as config from '@/api/mock-data/config.json'

const store = useStore()
function renderComponent(environment: string) {
  const clientConfig: ClientConfigInterface = { ...config, environment }
  store.state.config.clientConfig = clientConfig

  return mount(WelcomeView)
}

describe('WelcomeView.vue', () => {
  test('renders snapshot', () => {
    const wrapper = renderComponent('universal')

    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders Kubernetess', () => {
    const wrapper = renderComponent('kubernetess')

    expect(wrapper.html()).toContain('Kubernetes')
  })
})
