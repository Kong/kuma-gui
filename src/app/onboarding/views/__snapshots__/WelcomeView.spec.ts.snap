// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeView.vue renders snapshot 1`] = `
<div
  class="route-view"
>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="route-view-title visually-hidden"
  >
    Navigated to 
  </div>
  
  <!--v-if-->
  <main
    class="app-main-content"
  >
    
    <!--v-if-->
    <section
      class=""
    >
      <!--v-if-->
      
      <div>
        <div>
          <div
            class="onboarding-container"
          >
            <div
              class="onboarding-container__header"
            >
              
              <div
                class="onboarding-heading"
              >
                <h1
                  class="onboarding-title"
                >
                  
                   Welcome to Kuma
                  
                </h1>
                <div
                  class="onboarding-description"
                >
                  
                  <p>
                     Congratulations on downloading Kuma! You are just a 
                    <strong>
                      few minutes
                    </strong>
                     away from getting your service mesh fully online. 
                  </p>
                  <p>
                     We have automatically detected that you are running on 
                    <strong>
                      Universal
                    </strong>
                    . 
                  </p>
                  
                </div>
              </div>
              
            </div>
            <div
              class="onboarding-container__content"
            >
              <div
                class="onboarding-container__inner-content"
              >
                
                <h2
                  class="text-center"
                >
                   Letâ€™s get started: 
                </h2>
                <div
                  class="item-status-list-wrapper"
                >
                  <ul
                    class="item-status-list"
                  >
                    
                    <li>
                      <span
                        class="circle mr-2"
                      >
                        <span
                          class="kong-icon kong-icon-check"
                        >
                          <svg
                            fill="none"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            
  
                            <path
                              d="m14 7-6 6-3-3"
                              fill="none"
                              stroke="#A3BBCC"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="2.25"
                            />
                            

                          </svg>
                          

                        </span>
                      </span>
                       Run Kuma control plane
                    </li>
                    <li>
                      <span
                        class="circle mr-2"
                      >
                        <!--v-if-->
                      </span>
                       Learn about deployments
                    </li>
                    <li>
                      <span
                        class="circle mr-2"
                      >
                        <!--v-if-->
                      </span>
                       Learn about configuration storage
                    </li>
                    <li>
                      <span
                        class="circle mr-2"
                      >
                        <!--v-if-->
                      </span>
                       Create the mesh
                    </li>
                    <li>
                      <span
                        class="circle mr-2"
                      >
                        <!--v-if-->
                      </span>
                       Add services
                    </li>
                    <li>
                      <span
                        class="circle mr-2"
                      >
                        <!--v-if-->
                      </span>
                       Go to the dashboard
                    </li>
                    
                  </ul>
                </div>
                
              </div>
            </div>
            <div
              class="mt-4"
            >
              
              <div
                class="onboarding-actions"
              >
                <!--v-if-->
                <div
                  class="button-list"
                >
                  <a
                    class="k-button medium rounded outline"
                    data-testid="onboarding-skip-button"
                    href="/gui/"
                    type="button"
                  >
                    
                    <!---->
                    
                    
                     Skip setup 
                    
                    <!---->
                  </a>
                  <a
                    class="k-button medium rounded primary"
                    data-testid="onboarding-next-button"
                    href="/gui/onboarding/deployment-types"
                    type="button"
                  >
                    
                    <!---->
                    
                    
                    Next
                    
                    <!---->
                  </a>
                </div>
              </div>
              
            </div>
          </div>
          <div
            class="background-image"
          />
        </div>
        <svg
          class="background svg"
          longer="false"
          viewBox="0 0 1920 1080"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <defs>
            <lineargradient
              id="a"
              x1="64.8%"
              x2="34.4%"
              y1="-12.6%"
              y2="153.1%"
            >
              <stop
                offset="0%"
                stop-color="var(--onboarding-accent-dark)"
              />
              <stop
                offset="100%"
                stop-color="var(--onboarding-accent)"
              />
            </lineargradient>
            <lineargradient
              id="b"
              x1="0%"
              x2="107.4%"
              y1="41.5%"
              y2="41.5%"
            >
              <stop
                offset="0%"
                stop-color="var(--onboarding-accent-dark)"
              />
              <stop
                offset="100%"
                stop-color="var(--onboarding-accent)"
              />
            </lineargradient>
          </defs>
          <g
            fill="none"
            fill-rule="evenodd"
            mask="url(#ccard)"
            transform="translate(-63 -285)"
          >
            <g
              stroke="var(--onboarding-accent)"
              stroke-width="6"
            >
              <path
                d="M1444 893h252"
              />
              <path
                d="M1529 705h232M1452 603h237"
                stroke-opacity=".4"
              />
              <path
                d="M1754 563l-332 332h-76M1444 935l121 121M263 859l156 156"
              />
              <path
                d="M742 781H556"
                stroke-opacity=".4"
              />
              <path
                d="M697 736H513"
              />
              <path
                d="M695 783V577"
                stroke-opacity=".4"
              />
              <path
                d="M261 1026V751M509 573V438M1502 415l291 290"
              />
              <path
                d="M698 912L26 240M1368 411v540l61 61 95-95M1434 3h267l193 192v354"
                stroke-opacity=".4"
              />
              <path
                d="M517 411h342l138 138M1416 573v242l371 323"
              />
              <path
                d="M1486 817V612l-146-146M839 243h-97l-83 84v348M1698 1063V817l58-57h122M1069 299L558 810M696 1058H585L468 941V570L322 424"
              />
              <path
                d="M277 528l160 160 236-236 121 121M632 979h-45l-67-67v-86H0M106 669h275M70 707h331M207 745h210M85 784h356M1417 558h228M1609 634h203M528 946h76M619 604v131M1359 567l125 125M1332 594l156 156M1594 1070V959M381 632L260 753"
              />
              <path
                d="M605 817V497M1851 959h-518M944 570H390"
                stroke-opacity=".4"
              />
              <path
                d="M638 912H342M635 1139V912"
              />
            </g>
            <foreignobject
              height="100%"
              width="100%"
              x="0"
              y="0"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
              <div />
            </foreignobject>
          </g>
        </svg>
      </div>
      
    </section>
    
  </main>
  
</div>
`;
